actions/microsoft-onedrive/microsoft_onedrive/onedrive_get_action.py

```diff
@@ -0,0 +1,331 @@
+                        "@microsoft.graph.downloadUrl"
+                    )
+                else:
+                    item["download_url"] = None
+            else:
+                item["download_url"] = None
+
+        return Response(result={"items": filtered_items})
+    else:
+        raise ActionError(f"Failed to search items: {response.text}")
+
+
+@action
+def download_onedrive_file(
+    token: OAuth2Secret[
+        Literal["microsoft"],
+        List[Literal["Files.Read.All"]],
+    ],
+    download_request: DownloadRequest,
+    download_location: str = "",
+    download_all: bool = False,
+) -> Response[str]:
+    """
+    Download file or files from OneDrive to local storage.
+
+    Do not give 'download_location' unless user specifies a path.
+    By default downloads to the user's Downloads folder.
+
+    Args:
+        token: OAuth2 token to use for the operation.
+        download_request: The download request containing the search query or download URL.
+        download_location:  The path (folder) to save the downloaded file.
+        download_all: Whether to download all files found or not.
+
+    Returns:
+        Response with a message indicating the number of files downloaded.
+    """
+    download_location = (
+        Path.home() / "Downloads"
+        if download_location == ""
+        else Path(download_location)
+    )
+    if download_request.download_url and download_request.name:
+        matching_files = [
+            {
+                "name": download_request.name,
+                "download_url": download_request.download_url,
+            }
+        ]
+    elif download_request.search_items_request:
+        response = search_for_onedrive_items(
+            token,
+            download_request.search_items_request,
+        )
+        matching_files = response.result.get("items", [])
+        if len(matching_files) == 0:
+            raise ActionError("No files found matching the search query.")
+        elif not download_all and len(matching_files) > 1:
+            raise ActionError(
+                "Multiple files found matching the search query. Set download_all to True to download all files."
+            )
+
+    else:
+        raise ActionError("Either query or download_url, name must be provided.")
+    download_count = 0
+    for file in matching_files:
+        download_url = file.get("download_url")
+        if download_url:
+            file_name = file.get("name")
+            download_response = requests.get(download_url)
+            if download_response.status_code == 200:
+                with open(f"{download_location}/{file_name}", "wb") as f:
+                    f.write(download_response.content)
+                download_count += 1
+            else:
+                raise ActionError(f"Failed to download file: {download_response.text}")
+        else:
+            raise ActionError(f"No download URL available for file: {file.get('name')}")
```

Comment:
should we raise error here? or just log and move forward with all the rest of matching files instead of interrupting the execution?